src/Core/Utilities/CustomRedisProcessingStrategy.cs

```diff
@@ -0,0 +1,128 @@
+        IMemoryCache memoryCache,
+        GlobalSettings globalSettings)
+        : base(config)
+    {
+        _connectionMultiplexer = connectionMultiplexer ??
+                                 throw new ArgumentException(
+                                     "IConnectionMultiplexer was null. Ensure StackExchange.Redis was successfully registered");
+        _config = config;
+        _logger = logger;
+        _memoryCache = memoryCache;
+        _distributedSettings = globalSettings.DistributedIpRateLimiting;
+    }
+
+    private static readonly LuaScript _atomicIncrement = LuaScript.Prepare(
+        "local count = redis.call(\"INCRBYFLOAT\", @key, tonumber(@delta)) local ttl = redis.call(\"TTL\", @key) if ttl == -1 then redis.call(\"EXPIRE\", @key, @timeout) end return count");
+
+    public override async Task<RateLimitCounter> ProcessRequestAsync(ClientRequestIdentity requestIdentity,
+        RateLimitRule rule, ICounterKeyBuilder counterKeyBuilder, RateLimitOptions rateLimitOptions,
+        CancellationToken cancellationToken = default)
+    {
+        // If Redis is down entirely, skip rate limiting
+        if (!_connectionMultiplexer.IsConnected)
+        {
+            _logger.LogDebug("Redis connection is down, skipping IP rate limiting");
+            return SkipRateLimitResult();
+        }
+
+        // Check if any Redis timeouts have occured recently
+        if (_memoryCache.TryGetValue<TimeoutCounter>(_redisTimeoutCacheKey, out var timeoutCounter))
+        {
+            // We've exceeded threshold, backoff Redis and skip rate limiting for now
+            if (timeoutCounter.Count >= _distributedSettings.MaxRedisTimeoutsThreshold)
+            {
+                _logger.LogDebug(
+                    "Redis timeout threshold has been exceeded, backing off and skipping IP rate limiting");
+                return SkipRateLimitResult();
+            }
+        }
+
+        var counterId = BuildCounterKey(requestIdentity, rule, counterKeyBuilder, rateLimitOptions);
+
+        try
+        {
+            return await IncrementAsync(counterId, rule.PeriodTimespan ?? rule.Period.ToTimeSpan(), _config.RateIncrementer);
+        }
+        catch (RedisTimeoutException)
+        {
+            // If this is the first timeout we've had, start a new counter and sliding window 
+            timeoutCounter ??= new TimeoutCounter()
+            {
+                Count = 0,
+                ExpiresAt = DateTime.UtcNow.AddSeconds(_distributedSettings.SlidingWindowSeconds)
+            };
+            timeoutCounter.Count++;
+
+            _memoryCache.Set(_redisTimeoutCacheKey, timeoutCounter,
+                new MemoryCacheEntryOptions { AbsoluteExpiration = timeoutCounter.ExpiresAt });
+
+            // Just because Redis timed out does not mean we should kill the request
+            return SkipRateLimitResult();
+        }
+    }
+
+    private async Task<RateLimitCounter> IncrementAsync(string counterId, TimeSpan interval,
```

Comment:
Do you need this if we just call `base` like suggested above?

Reply:
Nope, we no longer need it if using `base`